src/main/java/io/cdap/plugin/gcp/gcs/actions/GCSBucketDelete.java

```diff
@@ -119,13 +128,80 @@ public void run(ActionContext context) throws Exception {
       if (fs.exists(gcsPath)) {
         try {
           fs.delete(gcsPath, true);
+          deleteCount++;
         } catch (IOException e) {
           LOG.warn(
             String.format("Failed to delete path '%s'", gcsPath)
           );
         }
       }
     }
+
+    for (GCSPath gcsPath : gcsPathsWild) {
+      String regex = getRegex(gcsPath);
+      try {
+        fs = new Path(gcsPath.getUri()).getFileSystem(configuration);
+      } catch (IOException e) {
+        LOG.info("Failed deleting file " + gcsPath.getUri().getPath() + ", " + e.getMessage());
+        // no-op.
+        continue;
+      }
+      Page<Blob> blobs = storage.list(gcsPathsWild.get(0).getBucket());
+      for (Blob blob : blobs.iterateAll()) {
+        Path filePath = new Path("gs://" + blob.getBucket() + "/" + blob.getName());
+        if (filePath.toString().matches(regex)) {
+          if (fs.exists(filePath)) {
+            Path rootPath = getRootMatch(filePath, regex);
+            try {
+              fs.delete(rootPath, true);
+              deleteCount++;
+            } catch (IOException e) {
+              LOG.warn(
+                String.format("Failed to delete path '%s'", gcsPath)
+              );
+            }
+          } else {
+            deleteCount++;
+          }
+        }
+      }
+    }
+    context.getMetrics().gauge("gc.file.delete.count", deleteCount);
+  }
+
+  public Path getRootMatch(Path filePath, String regx) {
+    Path parent = filePath;
+    String child = null;
+    while (parent.getParent() != null && parent.toString().matches(regx)) {
```

Comment:
try to simplify it by making parent as `current` and test `current.getParent().matches(regx)`